src/industry_gui.cpp

```diff
@@ -1390,8 +1553,17 @@ class IndustryDirectoryWindow : public Window {
 					DrawString(r.left + WD_FRAMERECT_LEFT, r.right - WD_FRAMERECT_RIGHT, y, STR_INDUSTRY_DIRECTORY_NONE);
 					break;
 				}
+				TextColour tc;
+				const CargoID acf_cid = this->cargo_filter[this->accepted_cargo_filter_criteria];
 				for (uint i = this->vscroll->GetPosition(); i < this->industries.size(); i++) {
-					DrawString(r.left + WD_FRAMERECT_LEFT, r.right - WD_FRAMERECT_RIGHT, y, this->GetIndustryString(this->industries[i]));
+					tc = TC_FROMSTRING;
+					if (acf_cid != CF_ANY && acf_cid != CF_NONE) {
+						Industry *ind = const_cast<Industry *>(this->industries[i]);
```

Comment:
instead of this cast, could `IndustryTemporarilyRefusesCargo` not have its signature changed? Would seem to me to be more "correct"

Reply:
I think to change the signature of  `IndustryTemporarilyRefusesCargo` then the `GetIndustryCallback` should be changed, and maybe other functions too. So I would not like to change it in this PR.

Reply:
I'd be happy with it in this PR as a separate commit, but am not fussed

Reply:
I tried to change it, but did not succesed. I always get error when trying to modify the `IndustriesScopeResolver`.